### ISSUE TITLE
Find source image by filters

### ISSUE THREAD

**Issue by konstantin-kornienko** [hashicorp/packer-plugin-alicloud#45]
#### Description

Plugin provides `source_image` option to find the base image in Alicloud. For the time being, we should hard-code image names or their ids. It's a pain.

It would be nice to have ability to specify filters to find image by name regex or tags, like in AWS plugin:
https://github.com/hashicorp/packer-plugin-amazon/blob/main/docs/datasources/ami.mdx

**Comment by virain** [comment]
I agree. The image ID changes frequently, which makes the original template need to be modified frequently. This is very painful

**Comment by alexyueer** [comment]
Alicloud  provide image_family for users to ignore frequently changed image ids,  the same option like source_image.
more details 
https://developer.hashicorp.com/packer/plugins/builders/alicloud
https://www.alibabacloud.com/help/en/elastic-compute-service/latest/image-family-overview

**Comment by akolkov-incountry** [comment]
@alexyueer thank, but in some cases it's not useful, because I can't set of modify image_family for new image, it's works only like source for build image. 
For example: I have base image Ubuntu_20_04 and I need update it, from time-to-time (security patches etc).
And I have some special images, with DB or some SDKs. I can't build this inherited images without hardcoding base image_id, because I can't set image_family attribute when I build latest base.

**Comment by alexyueer** [comment]
> @alexyueer thank, but in some cases it's not useful, because I can't set of modify image_family for new image, it's works only like source for build image. For example: I have base image Ubuntu_20_04 and I need update it, from time-to-time (security patches etc). And I have some special images, with DB or some SDKs. I can't build this inherited images without hardcoding base image_id, because I can't set image_family attribute when I build latest base.

Hi, sorry for the slow response. Add parameter like image_family_for_new to work in [CreateImage](https://www.alibabacloud.com/help/en/elastic-compute-service/latest/createimage), cloud it resolve this issue?

**Comment by konstantin-kornienko** [comment]
>>>dd parameter like image_family_for_new to work in [CreateImage](https://www.alibabacloud.com/help/en/elastic-compute-service/latest/createimage), cloud it resolve this issue?
@alexyueer, thanks for the response, but unfortunately it won't help...

The idea is to search for the image with filters and use the latest. Like:
```
data "alicloud-image" "basic-example" {
    filters = {[
        Name = "Name"
        Value = "*mt_base_image*"
    ]}
    owners = ["self"]
    most_recent = true
}
```

**Comment by niklas-nielsen** [comment]
I think adding a data source for alicloud_images similar to this Terraform datasource: https://registry.terraform.io/providers/aliyun/alicloud/latest/docs/data-sources/images to the plugin would be an elegant solution to this problem 👍

**Comment by michaelw730** [comment]
It would be best if the search mechanism was part of packer as suggested but as a workaround you could use python and the DescribeImagesRequest library (or another language) to search for the image and pass the image name in as a variable.
